src/go/rpk/pkg/cli/cmd/topic/produce.go

```diff
@@ -167,6 +171,7 @@ func newProduceCommand(fs afero.Fs) *cobra.Command {
 	cmd.Flags().IntVar(&acks, "acks", -1, "Number of acks required for producing (-1=all, 0=none, 1=leader)")
 	cmd.Flags().DurationVar(&timeout, "delivery-timeout", 0, "Per-record delivery timeout, if non-zero, min 1s")
 	cmd.Flags().Int32VarP(&partition, "partition", "p", -1, "Partition to directly produce to, if non-negative (also allows %p parsing to set partitions)")
+	cmd.Flags().Int32Var(&maxMessageBytes, "max-message-bytes", 1048576, "Maximum size of a record batch before compression")
```

Comment:
What about -1 to have an internal default (1,000,012) and then have this flag be used if it is non-negative?

```
If non-negative, maximum...
```
?

Reply:
Sounds reasonable. Thanks for the advise! I will file a follow-up PR.

Reply:
Created https://github.com/redpanda-data/redpanda/pull/7696